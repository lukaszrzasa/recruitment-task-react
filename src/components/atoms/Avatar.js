import styled from 'styled-components';
import PropTypes from 'prop-types';

export const sizes = {
  xs: 18,
  md: 30,
  xl: 55,
};

export const getSize = (size) => {
  if(Object.keys(sizes).indexOf(size)===-1) return sizes.md;
  return sizes[size];
};

const Avatar = styled.img`
  width: ${({size}) => getSize(size)}px;
  height: ${({size}) => getSize(size)}px;
  border-radius: 50%;
  display: inline-block;
`;

Avatar.propTypes = {
  size: PropTypes.oneOf(['xl', 'md', 'xs']),
};
Avatar.defaultProps = {
  size: 'md',
};

export default Avatar;