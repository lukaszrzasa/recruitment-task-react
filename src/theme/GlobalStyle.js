import { createGlobalStyle } from 'styled-components';

const GlobalStyle = createGlobalStyle`
  body, html {
    margin: 0;
    padding: 0;
    overflow: hidden;
  }
  * {
    overflow-wrap: break-word;
  }
`;

export default GlobalStyle;