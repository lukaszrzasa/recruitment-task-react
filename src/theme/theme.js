const colors = {
  violet: '#b33ed6',
  black: '#222',
  default: '#222',
  gray: '#809396',
  gold: '#ffdf13',
  white: '#fff',
};

const sizes = {
  xxl:200,
  xl: 60,
  lg:40,
  md:22,
  sm:18,
};

const theme = {
  colors,
  sizes,
};
export default theme;