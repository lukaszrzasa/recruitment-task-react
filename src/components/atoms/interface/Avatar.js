import styled from 'styled-components';
import PropTypes from 'prop-types';
import theme from '../../../theme/theme';
import getSize from '../helpers/getSize';

const { sizes } = theme;

const Avatar = styled.img`
  width: ${({size}) => getSize(size)}px;
  height: ${({size}) => getSize(size)}px;
  border-radius: 50%;
  display: inline-block;
`;

Avatar.propTypes = {
  size: PropTypes.oneOf(Object.keys(sizes)),
};
Avatar.defaultProps = {
  size: 'md',
};

export default Avatar;