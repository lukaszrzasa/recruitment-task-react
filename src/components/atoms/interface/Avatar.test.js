import React from 'react';
import {render} from '@testing-library/react';
import Avatar from './Avatar';
import theme from '../../../theme/theme';

const { sizes } = theme;

describe('atoms/Avatar component', ()=>{

  it('render properly',()=>{
    const { container } = render(<Avatar size="xs" />);
    expect(container.firstChild).toMatchSnapshot()
  });
  it('has good size',()=>{
    const { container, rerender } = render(<Avatar size="xs" />);
    expect( container.querySelector('img') ).toHaveStyle(`width: ${sizes.xs}px`);
    rerender(<Avatar size="md" />);
    expect( container.querySelector('img') ).toHaveStyle(`width: ${sizes.md}px`);
    rerender(<Avatar size="xl" />);
    expect( container.querySelector('img') ).toHaveStyle(`width: ${sizes.xl}px`);
  });

});