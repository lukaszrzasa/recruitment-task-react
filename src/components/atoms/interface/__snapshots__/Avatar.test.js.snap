// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms/Avatar component render properly 1`] = `
<img
  class="sc-AxjAm jbsMNI"
/>
`;
